pkg/localConfigProvider/localConfigProvider.go

```diff
@@ -63,7 +63,7 @@ type LocalConfigProvider interface {
 	ValidateURL(url LocalURL) error
 	CreateURL(url LocalURL) error
 	DeleteURL(name string) error
-	GetPorts() ([]string, error)
+	GetPorts(container string) ([]string, error)
```

Comment:
I think it would be easier to understand the behaviour of this function by splitting it into 2 functions:

- `GetContainerPorts(container string)`
- `GetComponentPorts()`

(I'm not sure about the naming of the second one)

The two functions could call a same private function handling the 2 cases if you want to not duplicate too much code.

Reply:
I guess something like that could be done. I just wanted to achieve this with minimal code change, but i guess refactoring stuff also makes sense.